specutils/utils/spectral_region.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import astropy.units as u
+
+
+class SpectralRegion:
+    """
+    SpectralRegion is currently a container class enables some simplicty
+    in defining and passing a region (interval) for a spectrum.
+
+    In the future, there might be more functionality added in here and there
+    is some discussion that this might/could move to astropy.region.
+    """
+
+    def __init__(self, lower, upper):
+        """
+        Lower and upper values for the interval.
+
+        Parameter
+        --------
+
+        lower: number with ``astropy.units`` unit
```

Comment:
Here (and in upper) I think you want to say:
```
lower : Scalar `~astropy.units.Quantity` with pixel or ``spectral_axis`` units
```
That is, a "number with ``astropy.units`` unit" *is* a `Quantity` 

Alternatively, you could do:
```
lower : Scalar `~astropy.units.Quantity` 
     The lower bound of the region. Should have units of `~astropy.units.pixel` or units that are convertable to match the `spectral_axis`.
```